src/mysql_protocol.erl

```diff
@@ -1259,6 +1266,23 @@ add_packet_headers_test() ->
     {Data, 43} = add_packet_headers(<<"foo">>, 42),
     ?assertEqual(<<3, 0, 0, 42, "foo">>, list_to_binary(Data)).
 
+add_packet_headers_equal_to_0xffffff_test() ->
+	BigBin = erlang:list_to_binary([16#01 || I <- lists:seq(1, 16#ffffff)]),
+    {Data, 43} = add_packet_headers(BigBin, 42),
+    ?assertEqual(<<16#ff, 16#ff, 16#ff, 42, BigBin/binary>>, list_to_binary(Data)).
```

Comment:
This is not the correct behaviour. A packet size of 16#ffffff means that one more packet will come. There is no other indicator of if there are more packets to come or not. Thus, a packet of size 0 has to follow. See https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_basic_packets.html .

In the implementation, it has to be `>=` in the guard, not `>`.

Reply:
Nice test cases by the way. I like them.

Reply:
We are also using custom patched version of mysql-otp purely because of limited packet size in upstream.

Reply:
Nice lib by the way. I like it :)

Reply:
Good to know that more people/projects need this. I would like to merge soon. (Sorry for being picky!) :-)